### PR TITLE
E2E-963/fi-happy-path

### PR DESCRIPTION
**Pull request by ololodream**
[E2E-963]
Happy path workflow

- Choose user 
- Login to FI
- Accept Terms
- Answer all questions (within length range)
- Submit assessment 
- Rate experience 
- Give feedback

### CODE REVIEW

File: cypress/support/actionTypes.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+export enum DATA_TYPES {
```
Comment (tonynguyenit18) [inline cypress/support/actionTypes.ts:1]: - I think we not using _ naming type for enum, I think we prefer PascalCase for it like `DataTypes`
- `DataTypes` is also too normal, it better to name it to `QuestionResponseType`

File: cypress/support/pageObjects/ChatBotPage.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  dropdownList(): Cypress.Chainable<JQuery<HTMLElement>> {
+    return cy.get(".option-list>.option-item")
+  }
+  submitButton(): Cypress.Chainable<JQuery<HTMLElement>> {
+    return cy.get(".submit-button-box")
+  }
+  mcqList(): Cypress.Chainable<JQuery<HTMLElement>> {
+    return cy.get(".option-list>.option-item")
+  }
+}
```
Comment (tonynguyenit18) [inline cypress/support/pageObjects/ChatBotPage.ts:29]: Should we create const for js class and id. e.g `const sendButtonClass = .ph-send-button`, doing this help to add comment sometime, for example we can create constant for `"#\\31"` then put the comment to let viewer know where the id comes from

**Review by tonynguyenit18 — COMMENTED** [review]
I have comments please have a look

File: cypress/support/actionTypes.ts (hunk 1 above)
Comment (ololodream) [reply cypress/support/actionTypes.ts:1]: Good one

File: cypress/support/pageObjects/ChatBotPage.ts (hunk 2 above)
Comment (ololodream) [reply cypress/support/pageObjects/ChatBotPage.ts:29]: I think the function name itself already tell what is this element, like `#\\31` is the selector for `acceptButton `

Comment (tonynguyenit18) [reply cypress/support/pageObjects/ChatBotPage.ts:29]: When I read this I'll question myself where the `#\\31`, what is that weird number.
If you can make it const you can add comment like "we are using xxx, it auto set the id for "accept button...."
For fixed string, it's always better to make it constant

Comment (ololodream) [reply cypress/support/pageObjects/ChatBotPage.ts:29]: I understand where are you from, but it is a little bit different in testing. PageObjects is kind of support functions, and it is the place to allocate elements. For example, you can call `chatBotPage. acceptButton()` to get accept button without taking care of how to find it. And pageObjects is the place to  creates an object repository for storing all web elements. It is useful in reducing code duplication and improves test case maintenance. Although we are using cypress, but we can still borrow this concept [Page Object Model and Page Factory in Selenium](https://www.browserstack.com/guide/page-object-model-in-selenium)

Comment (tonynguyenit18) [reply cypress/support/pageObjects/ChatBotPage.ts:29]: I'm not talking about the technical stuff, What I mentioned is if we have a fixed string,  we should better create constant for it with proper name, and incase of name can't not explain much you can easily add comments to explain more.
Above is the benefit of using variable for fixed string. I can't see any drawback of using it, why don't we use it?

**Review by tonynguyenit18 — APPROVED** [review]

File: cypress/support/pageObjects/ChatBotPage.ts (hunk 2 above)
Comment (tonynguyenit18) [reply cypress/support/pageObjects/ChatBotPage.ts:29]: ANW, it's just y suggestion, there is no problem if you wanna keep this. I approved.

**Pull request merged**